framework/src/base/MooseApp.C

```diff
@@ -1371,105 +1380,148 @@ MooseApp::run()
+    Moose::err << "Syntax OK" << std::endl;
+  }
+}
+
+bool
+MooseApp::copyInputs()
+{
+  std::string dir_to_copy = "tests";
+  bool copy_param_valid = false;
+
+  if (isParamValid("copy_inputs"))
+  {
+    // Get command line argument following --copy-inputs on command line
+    dir_to_copy = getParam<std::string>("copy_inputs");
+    copy_param_valid = true;
+  }
+
+  if (getParam<bool>("copy_tests") || copy_param_valid)
   {
     auto binname = appBinaryName();
     if (binname == "")
       mooseError("could not locate installed tests to run (unresolved binary/app name)");
-    auto src_dir = MooseUtils::installedTestsDir(binname);
+    auto src_dir = MooseUtils::installedInputsDir(binname, dir_to_copy);
     if (src_dir == "")
       mooseError("couldn't locate any installed tests to copy");
     if (!MooseUtils::checkFileReadable(src_dir, false, false))
       mooseError(
           "You don't have permissions to read/copy tests from their current installed location: \"",
           src_dir,
           "\"");
-    auto dst_dir = binname + "_tests";
+    auto dst_dir = binname + "_" + dir_to_copy;
     auto cmdname = Moose::getExecutableName();
     if (cmdname.find_first_of("/") != std::string::npos)
       cmdname = cmdname.substr(cmdname.find_first_of("/") + 1, std::string::npos);
     if (MooseUtils::pathExists(dst_dir))
-      mooseError("The tests directory \"./",
-                 dst_dir,
-                 "\" already exists.\nTo update/recopy tests, rename (\"mv ",
-                 dst_dir,
-                 " new_dir_name\") or remove (\"rm -r ",
-                 dst_dir,
-                 "\") the existing directory.\nThen re-run \"",
-                 cmdname,
-                 " --copy-tests\".");
+      mooseError(
+          "The directory \"./",
+          dst_dir,
+          "\" already exists.\nTo update/recopy the contents of this direcotry, rename (\"mv ",
+          dst_dir,
+          " new_dir_name\") or remove (\"rm -r ",
+          dst_dir,
+          "\") the existing directory.\nThen re-run \"",
+          cmdname,
+          " --copy-inputs ",
+          dir_to_copy,
+          "\".");
 
     std::string cmd = "cp -R " + src_dir + " " + dst_dir;
     int ret = system(cmd.c_str());
     if (WIFEXITED(ret) && WEXITSTATUS(ret) != 0)
-      mooseError("Failed to copy the tests.");
-    Moose::out << "Tests successfully copied into ./" << dst_dir << "\n";
-    _ready_to_exit = true;
-    return;
+      mooseError("Failed to copy the requested directory.");
+    Moose::out << "Directory successfully copied into ./" << dst_dir << "\n";
+    return true;
+  }
+  return false;
+}
+
+bool
+MooseApp::runInputs()
+{
+  std::string dir_to_run = "tests";
+  bool run_param_valid = false;
+  int skip_args = 2;
+
+  if (isParamValid("run"))
+  {
+    // Get command line argument following --run on command line
+    dir_to_run = getParam<std::string>("run");
+    run_param_valid = true;
+    skip_args = 3;
```

Comment:
is run always the first arg?

Reply:
Well it should be for running examples since it wouldn't make sense to send commands to MOOSE and run the TestHarness at the same time. If people do that now, and list run later, things are not going to work correctly. If they have MOOSE options after the run argument, they'll get passed to the TestHarness and that'll likely throw an error. This is all terrible logic that I am admittedly propagating. It can all be cleaned up.